### DISCUSSION TITLE
Automatically turn off runners if they have become idle for a period of time

### DISCUSSION THREAD

**Discussion by AllanOricil** [community/community#52822] [Actions]
### Select Topic Area

General

### Body

I would like to have a new configuration flag in the `config.sh` to enable the runner app to automatically shut down the runner when events have not been received after a certain amount of time.

````bash
config.sh .. --shutdown-on-idle
````

The runner app will signal the OS to shut it down when events have not been received in a period of N minutes.  This amount of time defaults to 30 minutes, but it is configurable using a variable called `SHUTDOWN_ON_IDLE_PERIOD=30`. Every time a new event is received, the timer resets. This means that the runner must wait agian 30 minutes before it is shut down.

With this configuration, customers won't need to create their own services to turn runners off. At the moment customers need to create a service to turn runners on and off. 
The service to turn runners on is quite easy to be done. Just need to create a lambda function that is triggered when receiveing events from a webhook. The lambda turns the runner on if it is stopped, otherwise just ignores it. You could also add an example of lambda for customers to copy and paste...

**Comment by mdellagi77** [comment]
Why has this been ignored and closed?

**Comment by Kamilcuk** [comment]
Hi, we stumbled upon we believe the same issue. The number of running we want to be dynamic, but we do not want to download big docker images every time we need to run it. `At the moment customers need to create a service to turn runners on and off` Have you implemented such your own service? Could you recommend any designs, share documentation or anything? Thank you!

**Reply by AllanOricil** [reply]
Tomorrow I can share my cdk project with you
